bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/local/TelldusLocalDeviceController.java

```diff
@@ -0,0 +1,287 @@
+            handleResponse((TellstickLocalDevice) dev, response);
+        } else {
+            throw new TelldusBindingException("Cannot send ON to " + dev);
+        }
+    }
+
+    @Override
+    public State calcState(Device dev) {
+        TellstickLocalDevice device = (TellstickLocalDevice) dev;
+        State st = null;
+
+        switch (device.getState()) {
+            case JNA.CLibrary.TELLSTICK_TURNON:
+                st = OnOffType.ON;
+                break;
+            case JNA.CLibrary.TELLSTICK_TURNOFF:
+                st = OnOffType.OFF;
+                break;
+            case JNA.CLibrary.TELLSTICK_DIM:
+                BigDecimal dimValue = new BigDecimal(device.getStatevalue());
+                if (dimValue.intValue() == 0) {
+                    st = OnOffType.OFF;
+                } else if (dimValue.intValue() >= 255) {
+                    st = OnOffType.ON;
+                } else {
+                    st = OnOffType.ON;
+                }
+                break;
+            default:
+                logger.warn("Could not handle {} for {}", device.getState(), device);
+        }
+
+        return st;
+    }
+
+    @Override
+    public BigDecimal calcDimValue(Device device) {
+        BigDecimal dimValue = new BigDecimal(0);
+        switch (((TellstickLocalDevice) device).getState()) {
+            case JNA.CLibrary.TELLSTICK_TURNON:
+                dimValue = new BigDecimal(100);
+                break;
+            case JNA.CLibrary.TELLSTICK_TURNOFF:
+                break;
+            case JNA.CLibrary.TELLSTICK_DIM:
+                dimValue = new BigDecimal(((TellstickLocalDevice) device).getStatevalue());
+                dimValue = dimValue.multiply(new BigDecimal(100));
+                dimValue = dimValue.divide(new BigDecimal(255), 0, BigDecimal.ROUND_HALF_UP);
+                break;
+            default:
+                logger.warn("Could not handle {} for {}", (((TellstickLocalDevice) device).getState()), device);
+        }
+        return dimValue;
+    }
+
+    public long getLastSend() {
+        return lastSend;
+    }
+
+    public void setLastSend(long currentTimeMillis) {
+        lastSend = currentTimeMillis;
+    }
+
+    @Override
+    public void onRequest(TellstickSensorEvent newDevices) {
+        setLastSend(newDevices.getTimestamp());
+    }
+
+    @Override
+    public void onRequest(TellstickDeviceEvent newDevices) {
+        setLastSend(newDevices.getTimestamp());
+    }
+
+    <T> T callRestMethod(String uri, Class<T> response) throws TelldusLocalException {
+        T resultObj = null;
+        try {
+            for (int i = 0; i < MAX_RETRIES; i++) {
+                try {
+                    resultObj = innerCallRest(localApiUrl + uri, response);
+                    break;
+                } catch (TimeoutException e) {
+                    logger.warn("TimeoutException error in get", e);
+                } catch (InterruptedException e) {
```

Comment:
The `InterruptedException` is thrown when OH is shut-down. That means the current operation should be finished as fast as possible and no further operations should be done. You can achieve this by not catching the exception here but throw the InterruptedException and catch it at the highest level you can. As this is not an error, it shouldn't be logged, but only handled.

Reply:
Added throws and moved catch to highest possible level.